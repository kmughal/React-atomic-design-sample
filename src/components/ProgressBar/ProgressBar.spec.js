import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ProgressBar from "./ProgressBar";
import renderer from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() });

describe("ProgressBar", () => {
	test("width 500 with percentage 20% should return 100", () => {
		const wrapper = Enzyme.shallow(
			<ProgressBar percent={20} width={500} height={100} />
		);
		const width = wrapper.instance().getWidthAsPercentInTotalWidth();
		expect(width).toEqual(100);
	});
});

describe("ProgressBar - snapshot testing", () => {
  test("should render a progress bar",()=>{
    const tree = renderer
		.create(<ProgressBar percent={20} width={500} height={100} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
});


// it is silly test but just have three levels covered!

describe("ProgressBar interaction testing example" ,()=> {
  test("should have a div" ,()=> {
    const wrapper = Enzyme.shallow(
			<ProgressBar percent={20} width={500} height={100} />
    );
    expect(wrapper.find('div')).toHaveLength(2)
  })
})