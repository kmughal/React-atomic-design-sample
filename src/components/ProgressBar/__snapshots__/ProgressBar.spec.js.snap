// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  style={
    Object {
      "border": "1px solid gray",
      "width": 500,
    }
  }
>
  <div
    style={
      Object {
        "background": "red",
        "height": 100,
        "width": 100,
      }
    }
  />
</div>
`;

exports[`ProgressBar - snapshot testing should render a progress bar 1`] = `
<div
  style={
    Object {
      "border": "1px solid gray",
      "width": 500,
    }
  }
>
  <div
    style={
      Object {
        "background": "red",
        "height": 100,
        "width": 100,
      }
    }
  />
</div>
`;
